composer/core/types.py

```diff
@@ -33,7 +33,7 @@
     from typing import Protocol
 
 __all__ = [
-    "Batch", "BatchPair", "BatchDict", "PyTorchScheduler", "JSON", "MemoryFormat", "as_batch_dict", "as_batch_pair",
+    "Batch", "BatchPair", "BatchDict", "PyTorchScheduler", "JSON", "MemoryFormat",
```

Comment:
Can you remove "Batch", "BatchPair", and "BatchDict" from `__all__`, and also remove these variables from lines 42-44 below?

Reply:
Ok I removed `BatchPair` and `BatchDict` in 
0545b64, but I guess removing `Batch` will be a bit tougher. I guess find-replace `Batch` -> `torch.Tensor`?

Reply:
I think `Batch` would be `Any`? Although this might unleash pyright hell...

Reply:
In the ideal world, we could use a `TypeVar` for the batch type. But I think this will introduce more pyright hell than it is worth -- especially for algorithms -- as they would likely be `cast`ing the batch anyways to silence pyright. So agreed; `Any` is the way to go.

Reply:
+1 for Any. I tried using typevar for this and was immediately buried in pyright fire and brimstone.

Reply:
ok fixed in [acf116c](https://github.com/mosaicml/composer/pull/954/commits/acf116c968f0458ddffd28ab9a4ad25b98c7b182)